sycl/source/detail/platform_impl.hpp

```diff
@@ -180,12 +182,16 @@ class platform_impl {
   static std::shared_ptr<platform_impl>
   getPlatformFromPiDevice(RT::PiDevice PiDevice, const plugin &Plugin);
 
+  context getDefaultContext();
+
 private:
   bool MHostPlatform = false;
   RT::PiPlatform MPlatform = 0;
   std::shared_ptr<plugin> MPlugin;
   std::vector<std::weak_ptr<device_impl>> MDeviceCache;
   std::mutex MDeviceMapMutex;
+  std::shared_ptr<context> MDefaultContext;
```

Comment:
`context` is a wrapper around `std::shared_ptr<context_impl>`. So, suggest storing shared_ptr to context_impl directly here. And create a `context` from it by calling `detail::createSyclObjFromImpl<context>` like in source/backend.cpp:76 .